src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -999,25 +998,17 @@ public void dropScopedCredentials(List<String> credentialNames) {
     cleanup(credentialNames, dropScopedCredentialTemplate);
   }
 
-  public List<Map<String, Optional<Object>>> getSnapshotTableData(
-      AuthenticatedUserRequest userRequest,
-      Snapshot snapshot,
+  public List<Map<String, Optional<Object>>> getTableData(
+      Table table,
       String tableName,
+      String datasetSourceName,
+      String parquetFileLocation,
       int limit,
       int offset,
       String sort,
       SqlSortDirection direction,
       String filter) {
 
-    // Ensure that the table is a valid table
-    SnapshotTable table =
-        snapshot
-            .getTableByName(tableName)
-            .orElseThrow(
-                () ->
-                    new InvalidTableException(
-                        "Table %s was not found in the snapshot".formatted(tableName)));
-
     // Ensure that the sort column is a valid column
```

Comment:
Removed this check in order to share the getTableData method between Azure snapshots and datasets. I found that this check was already performed in both the [SnapshotService](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/snapshot/SnapshotService.java#L764) and the [DatasetService](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/dataset/DatasetService.java#L501) within the respective retrieve preview and retrieve data endpoints.